scaaml/stats/print_stats.py

```diff
@@ -80,13 +81,14 @@ def __init__(self, measurements_info: Dict,
                 for trace_name in measurements_info
             } for split in Dataset.SPLITS
         }
-        self._split_trace_group_stats = {
-            split: {
-                trace_name: defaultdict(self._all_trace_stats)
-                for trace_name in measurements_info
-            } for split in Dataset.SPLITS
-        }
-        self._all_groups = set()
+        self._split_trace_group_stats: Dict[str, Dict[str, Dict[
```

Comment:
Not to be part of this PR but nested data structures aren't the most readable thing in Python.
Maybe consider using `dataclasses` here?

Reply:
https://github.com/google/scaaml/issues/74